### ISSUE TITLE
Same `type` for 2 resources in two different namespaces throws `Uncaught Error: Type "<some-type>" already registered on base class

### ISSUE THREAD

**Issue by andhapp** [graphiti-api/spraypaint.js#48]
I have 2 resources in different namespaces with the same base url. When I try and set the `jsonapiType` for both of these to the same string - `my_type` in my models, I receive:

```
Uncaught Error: Type "<my_type>" already registered on base class function Subclass() {
                return _super !== null && _super.apply(this, arguments) || this;
```

I had a little look and I think it's down to [`JsonapiTypeRegistry` class](https://github.com/graphiti-api/spraypaint.js/blob/803721f1b4959dbb31b3b66bc2a35da720eb5374/src/jsonapi-type-registry.ts#L11-L16). 

Will be happy to submit a PR.

**Comment by richmolj** [comment]
You should be able to handle this with a separate `ApplicationResource` - each one gets its own type registry. So If you had `ModelA` inherit from `ApplicationResourceA`, and `ModelB` inherit from `ApplicationResourceB`, it should work. Let me know how it goes!

**Comment by richmolj** [comment]
Yes, both models should work...that said, I'm not sure if the *relationship* would work.

**Comment by andhapp** [comment]
That solution worked! Thanks @richmolj.

Still feels like a workaround and given that we are unsure about the **_relationships_**, do you think this could be solved in another way?

Again, happy to put the work in to create a PR.

**Comment by richmolj** [comment]
So, a little background on the thinking here. The id/type combo is meant to uniquely identify records. So let's imagine `API1` has employees and tags, `API1` has positions and tags.

Imagine a request like "load the employee, their tags, their positions, and the tags in those positions". Our API response could have the resource identifier `tags/123` in two places - we wouldn't be able to uniquely identify records.

Spraypaint works by building a registry of `Model: type`. This way when we look at a response, we don't need to do anything more than say "find the corresponding model for this type and instantiate it". If there are multiple types, we wouldn't be able to do this. It's especially a concern where the client isn't making a request (imagine pushing a JSON:API payload over a websocket, and needing to find the corresponding model instances in memory).

So I think it runs against the JSON:API grain a bit. That said, it's not uncommon and you're not the only user that's asked about this. I typically say that you should prefix the types to be unique, e.g. `api2/tags`. That said, I'm not against a PR that allows multiple types (if such a thing is possible), but it's likely a significant refactor.
